### PR TITLE
CMake: Enable CXX compiler for tests only

### PR DESCRIPTION
**Pull request by gadoofou87**
Fixes https://github.com/google/cpu_features/issues/109

### CODE REVIEW

**Comment by Mizux** [comment]
LGTM, just need to wait for CI to finish

**Comment by gadoofou87** [comment]
I think the build error is due to this commit https://github.com/google/googletest/commit/23b2a3b1cf803999fb38175f6e9e038a4495c8a5

**Comment by Mizux** [comment]
I think we should set properties on source files to set LANGUAGE to CXX
ref: https://cmake.org/cmake/help/latest/prop_sf/LANGUAGE.html

File: CMakeLists.txt [hunk 1]
```diff
@@ -197,6 +197,7 @@ if(BUILD_TESTING)
                      EXCLUDE_FROM_ALL)
   endif()
 
+  enable_language(CXX)
```
Comment (Mizux) [inline CMakeLists.txt]: you should try to move it just after `if(BUILD_TESTING)` line 162 so googletest is build with support enabled

**Comment by gadoofou87** [comment]
Almost fixed! :)

**Comment by Mizux** [comment]
Last remaining issue is ci setup bug (no ninja on ppc64) -> unrelated to this PR

**Comment by Mizux** [comment]
BTW I was not able to reproduce it using a docker
ci/Dockerfile:
```Dockerfile
# Create a virtual environment with all tools installed
# ref: https://hub.docker.com/_/ubuntu
FROM ubuntu:rolling AS env
LABEL maintainer="mizux.dev@gmail.com"
# Install system build dependencies
RUN apt-get update -qq \
&& apt-get install -yq git wget curl libssl-dev build-essential ninja-build \
&& apt-get install -yq python3-dev python3-pip \
&& apt-get clean \
&& rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
# Install CMake 3.16.4
RUN wget "https://cmake.org/files/v3.16/cmake-3.16.4-Linux-x86_64.sh" \
&& chmod a+x cmake-3.16.4-Linux-x86_64.sh \
&& ./cmake-3.16.4-Linux-x86_64.sh --prefix=/usr/local/ --skip-license \
&& rm cmake-3.16.4-Linux-x86_64.sh
CMD [ "/bin/bash" ]

# Create lib directory
WORKDIR /home/lib
# Bundle lib source
COPY . .

FROM env as build
RUN cmake -S. -Bbuild -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTING=ON -GNinja
#RUN cmake -S. -Bbuild -DBUILD_TESTING=ON
RUN cmake --build build

FROM build as test
RUN cmake --build build --target test
```

```sh
docker build -t cpu_feature  -f ci/Dockerfile  .
```

**Comment by Mizux** [comment]
PYI last master build was one month ago, without ppc64 jobs
see: https://travis-ci.org/github/google/cpu_features/builds/646766654?utm_medium=notification&utm_source=github_status

I've just triggered a build master: https://travis-ci.org/github/google/cpu_features/builds/661073354?utm_medium=notification&utm_source=github_status

**Pull request merged**